mocha_test/groupBy.js

```diff
@@ -0,0 +1,375 @@
+var async = require('../lib');
+var expect = require('chai').expect;
+var assert = require('assert');
+
+describe('groupBy', function() {
+    this.timeout(250);
+
+    function groupByIteratee(callOrder, val, next) {
+        setTimeout(function() {
+            callOrder.push(val);
+            next(null, val+1);
+        }, val * 25);
+    }
+
+    context('groupBy', function() {
```

Comment:
Yes, I was looking at the [mapValues tests](https://github.com/caolan/async/blob/master/mocha_test/mapValues.js) when I added that. We use it in the `compose` and `queue` tests too.

It's more personal preference than anything, but I feel like it makes the test results easier to read and it forces them to be more organized:
![image](https://cloud.githubusercontent.com/assets/7647696/22677555/6a9f93a8-ecc1-11e6-91df-82b410cd04ab.png)